src/main/java/org/symphonyoss/integration/webhook/github/parser/v2/GithubMetadataParser.java

```diff
@@ -0,0 +1,210 @@
+import com.google.common.cache.LoadingCache;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.symphonyoss.integration.model.message.Message;
+import org.symphonyoss.integration.model.yaml.IntegrationProperties;
+import org.symphonyoss.integration.service.UserService;
+import org.symphonyoss.integration.webhook.github.parser.GithubParser;
+import org.symphonyoss.integration.webhook.github.parser.GithubParserException;
+import org.symphonyoss.integration.webhook.github.parser.GithubParserUtils;
+import org.symphonyoss.integration.webhook.parser.metadata.MetadataParser;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.ws.rs.ProcessingException;
+
+/**
+ * Abstract Github parser responsible to augment the Github input data querying the user API and
+ * pre-processing the input data.
+ * Created by campidelli on 02/05/17.
+ */
+public abstract class GithubMetadataParser extends MetadataParser implements GithubParser {
+
+  private IntegrationProperties integrationProperties;
+
+  private static final String PATH_IMG_ICON = "img/github_logo.svg";
+
+  private static final String INTEGRATION_NAME = "github";
+
+  private static final Logger LOG = LoggerFactory.getLogger(GithubMetadataParser.class);
+
+  private static final int CACHE_EXPIRATION_IN_MINUTES = 60;
+
+  private UserService userService;
+
+  private String integrationUser;
+
+  private GithubParserUtils utils;
+
+  private LoadingCache<String, String> userServiceInfoCache;
+
+  @Autowired
+  public GithubMetadataParser(UserService userService, GithubParserUtils utils,
+      IntegrationProperties integrationProperties) {
+    this.userService = userService;
+    this.utils = utils;
+    this.integrationProperties = integrationProperties;
+  }
+
+  @Override
+  public void setIntegrationUser(String integrationUser) {
+    this.integrationUser = integrationUser;
+  }
+
+  @Override
+  public Message parse(Map<String, String> parameters, JsonNode node) throws GithubParserException {
+    Message message = parse(node);
+    if (message != null && message.getData() != null) {
+      String data = message.getData().replace("\\n", MESSAGEML_LINEBREAK);
+      message.setData(data);
+    }
+    return message;
+  }
+
+  @Override
+  public Message parse(Map<String, String> headers, Map<String, String> parameters, JsonNode node)
+      throws GithubParserException {
+    addTagToNode(node, EVENT_TAG, headers.get(GITHUB_HEADER_EVENT_NAME));
+    return parse(parameters, node);
+  }
+
+  @PostConstruct
+  @Override
+  public void init() {
+    super.init();
+    userServiceInfoCache = CacheBuilder.newBuilder().expireAfterWrite(CACHE_EXPIRATION_IN_MINUTES,
```

Comment:
You wouldn't need to override this callback if you instantiate the cache in the constructor.